core/error/error_list.cpp

```diff
@@ -30,6 +30,8 @@
 
 #include "error_list.h"
 
+#include <array>
```

Comment:
It should not make any practical difference, but I would use a more "neutral" header that is already included in some parts of the code, like `<iterator>. Including `<array>` might suggest that we are using `std::array`, which we do not.

Reply:
Good call! I thought `<array>` is the most 'neutral', but I forgot `std::array` was even a thing 😅